specification/authorization/resource-manager/Microsoft.Authorization/preview/2018-07-01-preview/authorization-DenyAssignmentGetCalls.json

```diff
@@ -0,0 +1,494 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "AuthorizationManagementClient",
+    "version": "2018-07-01-preview",
+    "description": "Role based access control provides you a way to apply granular level policy administration down to individual resources or resource groups. These operations enable you to get deny assignments. A deny assignment describes the set of actions on resources that are denied for Azure Active Directory users."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{parentResourcePath}/{resourceType}/{resourceName}/providers/Microsoft.Authorization/denyAssignments": {
+      "get": {
+        "tags": [
+          "DenyAssignments"
+        ],
+        "operationId": "DenyAssignments_ListForResource",
+        "description": "Gets deny assignments for a resource.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceProviderNamespaceParameter"
+          },
+          {
+            "name": "parentResourcePath",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The parent resource identity.",
+            "x-ms-skip-url-encoding": true
+          },
+          {
+            "name": "resourceType",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The resource type of the resource.",
+            "x-ms-skip-url-encoding": true
+          },
+          {
+            "name": "resourceName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource to get deny assignments for."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/FilterParameter"
+          }
+        ],
+        "responses": {
+          "200": {
```

Comment:
Please consider adding `default` response to all our operations - [https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/creating-swagger.md#default-response](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/creating-swagger.md#default-response) . Defining Error object and using it in the default responses will add better error handling in generated code. Users will know what to expect and how to check error response codes. Otherwise e.g. C# will just try to deserialize error message to CloudError and may simply truncate some additional data [https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/Authorization/Management.Authorization/Generated/ClassicAdministratorsOperations.cs#L169](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/Authorization/Management.Authorization/Generated/ClassicAdministratorsOperations.cs#L169)

Reply:
The existing APIs don't specify a default response and we want to model the new APIs behavior to be similar to the existing ones.